src/axom/quest/examples/quest_candidates_example.cpp

```diff
@@ -0,0 +1,754 @@
+
+  // Get ids of necessary allocators
+  const int host_allocator =
+    axom::getUmpireResourceAllocatorID(umpire::resource::Host);
+  const int kernel_allocator = on_device
+    ? axom::getUmpireResourceAllocatorID(umpire::resource::Device)
+    : axom::execution_space<ExecSpace>::allocatorID();
+
+  // Copy the insert hexes to the device, if necessary
+  // Either way, insert_hexes_v will be a view w/ data in the correct space
+  auto& insert_hexes_h = insertMesh.hexes();
+  HexArray insert_hexes_d =
+    on_device ? HexArray(insert_hexes_h, kernel_allocator) : HexArray();
+
+  // Copy the insert bboxes to the device, if necessary
+  // Either way, insert_bbox_v will be a view w/ data in the correct space
+  auto& insert_bbox_h = insertMesh.hexBoundingBoxes();
+  BBoxArray insert_bbox_d =
+    on_device ? BBoxArray(insert_bbox_h, kernel_allocator) : BBoxArray();
+  auto insert_bbox_v = on_device ? insert_bbox_d.view() : insert_bbox_h.view();
+
+  // Bounding box of entire insert mesh
+  HexMesh::BoundingBox insert_mesh_bbox_h = insertMesh.meshBoundingBox();
+
+  // Copy the query hexes to the device, if necessary
+  // Either way, query_hexes_v will be a view w/ data in the correct space
+  auto& query_hexes_h = queryMesh.hexes();
+  HexArray query_hexes_d =
+    on_device ? HexArray(query_hexes_h, kernel_allocator) : HexArray();
+
+  // Copy the query bboxes to the device, if necessary
+  // Either way, bbox_v will be a view w/ data in the correct space
+  auto& query_bbox_h = queryMesh.hexBoundingBoxes();
+  BBoxArray query_bbox_d =
+    on_device ? BBoxArray(query_bbox_h, kernel_allocator) : BBoxArray();
+  auto query_bbox_v = on_device ? query_bbox_d.view() : query_bbox_h.view();
+
+  axom::utilities::Timer timer;
+  timer.start();
+
+  // If given resolution is less than one, use the cube root of the
+  // number of hexes
+  if(resolution < 1)
+  {
+    resolution = (int)(1 + std::pow(insertMesh.numHexes(), 1 / 3.));
+  }
+
+  const axom::primal::Point<int, 3> resolutions(resolution);
+
+  axom::spin::ImplicitGrid<3, ExecSpace, int> gridIndex(insert_mesh_bbox_h,
+                                                        &resolutions,
+                                                        insertMesh.numHexes(),
+                                                        kernel_allocator);
+  gridIndex.insert(insertMesh.numHexes(), insert_bbox_v.data());
+  timer.stop();
+  SLIC_INFO(
+    axom::fmt::format("0: Initializing Implicit Grid took {:4.3} seconds.",
+                      timer.elapsedTimeInSec()));
+
+  timer.start();
+  IndexArray offsets_d(query_bbox_v.size(), query_bbox_v.size(), kernel_allocator);
+  IndexArray counts_d(query_bbox_v.size(), query_bbox_v.size(), kernel_allocator);
+  IndexArray candidates_d(0, 0, kernel_allocator);
+
+  auto offsets_v = offsets_d.view();
+  auto counts_v = counts_d.view();
+
+  gridIndex.getCandidatesAsArray(query_bbox_v, offsets_v, counts_v, candidates_d);
```

Comment:
Is there a way to pass in a lambda to also check actual bounding boxes before marking candidates? I'd expect that to significantly improve the ImplciitGrid results.

I think we have some examples that do this in axom.